### PR TITLE
Add oVirt users/groups and users_facts/group/facts modules

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

ovirt_groups
ovirt_groups_facts
ovirt_users
ovirt_users_facts
##### ANSIBLE VERSION

```
ansible 2.1.2.0
```
##### SUMMARY

This PR adds new oVirt modules for managing users and groups in oVirt and also add modules for retriving facts about groups and users.

```

```

### CODE REVIEW

**Comment by machacekondra** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @machacekondra for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
@mwperina @ryansb

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_groups_facts.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search group X use following pattern: name=X"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all groups stars on string C<admin*>:
```
Comment (mwperina) [inline cloud/ovirt/ovirt_groups_facts.py]: I'd rephrase to: 'Gather facts about all groups which names start with C<admin>:'

File: cloud/ovirt/ovirt_users.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+
+class UsersModule(BaseModule):
+
+    def build_entity(self):
+        return otypes.User(
+            domain=otypes.Domain(
+                name=self._module.params['authz_name']
+            ),
+            user_name='{}@{}'.format(
```
Comment (mwperina) [inline cloud/ovirt/ovirt_users.py]: Please extract username formatting into a method and reuse this method.

File: cloud/ovirt/ovirt_users.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            connection=connection,
+            module=module,
+            service=users_service,
+        )
+
+        state = module.params['state']
+        if state == 'present':
+            ret = users_module.create(
+                search_params={
+                    'usrname': '{}@{}'.format(
```
Comment (mwperina) [inline cloud/ovirt/ovirt_users.py]: Please extract username formatting into a method and reuse this method.

File: cloud/ovirt/ovirt_users.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                        module.params['name'],
+                        module.params['authz_name']
+                    )
+                }
+            )
+        elif state == 'absent':
+            ret = users_module.remove(
+                search_params={
```
Comment (mwperina) [inline cloud/ovirt/ovirt_users.py:152]: Please extract username formatting into a method and reuse this method.

File: cloud/ovirt/ovirt_users_facts.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search user X use following pattern: name=X"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all users stars on string C<john*>:
```
Comment (mwperina) [inline cloud/ovirt/ovirt_users_facts.py]: I'd rephrase to: 'Gather facts about all users which first names start with C<john>:'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_groups_facts.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search group X use following pattern: name=X"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all groups which names start with C(admin*):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_groups_facts.py]: 'C(admin*)' -> 'C(admin)'

File: cloud/ovirt/ovirt_users_facts.py (hunk 5 above)
Comment (mwperina) [reply cloud/ovirt/ovirt_users_facts.py]: I'd rephrase to: 'Gather facts about all users which first names start with C(john):'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_users_facts.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search user X use following pattern: name=X"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all users stars on string C<john>:
```
Comment (mwperina) [inline cloud/ovirt/ovirt_users_facts.py]: Please rephrase: 'Gather facts about all users which names start with C(john):'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_users_facts.py (hunk 7 above)
Comment (machacekondra) [reply cloud/ovirt/ovirt_users_facts.py]: done

File: cloud/ovirt/ovirt_users_facts.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search user X use following pattern: name=X"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all users stars on string C(john):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_users_facts.py]: Please rephrase to: 'Gather facts about all users which first names start with C(john):'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_users_facts.py (hunk 8 above)
Comment (machacekondra) [reply cloud/ovirt/ovirt_users_facts.py]: done

**Comment by machacekondra** [comment]
shipit

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! :+1:

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
shipit

**Review by mwperina — APPROVED** [review]
:+1:
